### ISSUE TITLE
SQL Server Bug after 2.3 Upgrade - [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]JSON text is not properly formatted.

### ISSUE THREAD

**Issue by pbutler312** [hasura/graphql-engine#8319]
### Version Information

Server Version: 2.3

### Environment

OSS

### What is the expected behaviour?

200, data returned successfully

### What is the current behaviour?

After upgrading to version 2.3 from 2.0.5, we started receiving the following error when we would try to get one of our nested attributes: 
```
[Microsoft][ODBC Driver 17 for SQL Server][SQL Server]JSON text is not properly formatted. Unexpected character '.' is found at position 4.
```

Please refer to this stack overflow for a more detailed example:
https://stackoverflow.com/questions/70572476/hasura-query-that-works-similar-to-a-sql-left-join

We fixed the issue by going back down to 2.0.5, but this caused a large outage for us.

**Comment by MareoRaft** [comment]
I am also affected by this bug.

**Comment by dennislaviolette** [comment]
Thanks for raising the issue @pbutler312 and @MareoRaft. We are placing into our triage list and should be able to reach back out soon with next steps.

**Comment by sassela** [comment]
Thank you for the bug report and your patience, @pbutler312 and @MareoRaft. This bug was fixed at https://github.com/hasura/graphql-engine/commit/109a0beca80625084d04ecfcb5327671bde8f0df, is currently available in release [v2.4.0-beta.1 (pre-release)](https://github.com/hasura/graphql-engine/releases/tag/v2.4.0-beta.1) onwards and will be included in the next stable release within two weeks.

I'll close this issue for now, but please feel free to comment here if you have any questions.
